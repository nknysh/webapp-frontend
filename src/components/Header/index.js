// Libraries
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

// App
import { colors } from 'styles';
import { getCurrentUser } from 'store/modules/auth/selectors';

// Components
import Styled from '../Styled';
import UserDropdown from '../UserDropdown';

const Container = Styled.View.extend`
  background-color: ${colors.white16};
  z-index: 100;
`;

const Content = Styled.View.extend`
  height: 100px;
  width: 1160px;
  flex-direction: row;
  align-self: center;
  align-items: center;
  justify-content: space-between;
`;

const Logo = Styled.View.extend`
  width: 200px;
  background-color: ${colors.gray11};
`;

const Links = Styled.View.extend`
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Text = Styled.H7.extend`
  margin-horizontal: 10px;
  color: ${colors.gold10};
`;

const isAdmin = user => user && user.type === 'admin';
const isSalesRepresentative = user => user && user.type === 'sr';
const isRatesAdmin = user => user && user.type === 'rates-admin';

const Header = ({ currentUser }) => (
  <Container>
    <Content>
      <Logo />
      <Links>
        {isAdmin(currentUser) ? (
          <React.Fragment>
            <Link to="/">
              <Text>HOME</Text>
            </Link>
            <Link to="/admin/dashboard">
              <Text>DASHBOARD</Text>
            </Link>
            <Link to="/messages">
              <Text>MESSAGES</Text>
            </Link>
            <Link to="/admin/hotels">
              <Text>HOTELS</Text>
            </Link>
            <Link to="/admin/bookings">
              <Text>BOOKINGS</Text>
            </Link>
            <Link to="/admin/proposals">
              <Text>PROPOSALS</Text>
            </Link>
            <Link to="/admin/travel-agents">
              <Text>TRAVEL AGENTS</Text>
            </Link>
            <Link to="/admin/sales-representatives">
              <Text>SALES REPS</Text>
            </Link>
          </React.Fragment>
        ) : isSalesRepresentative(currentUser) ? (
          <React.Fragment>
            <Link to="/">
              <Text>HOME</Text>
            </Link>
            <Link to="/messages">
              <Text>MESSAGES</Text>
            </Link>
            <Link to="/calendar">
              <Text>CALENDAR</Text>
            </Link>
            <Link to="/proposals">
              <Text>PROPOSALS</Text>
            </Link>
            <Link to="/bookings">
              <Text>BOOKINGS</Text>
            </Link>
            <Link to="/holds">
              <Text>HOLDS</Text>
            </Link>
            <Link to="/admin/travel-agents">
              <Text>TRAVEL AGENTS</Text>
            </Link>
          </React.Fragment>
        ) : isRatesAdmin(currentUser) ? (
          <React.Fragment>
            <Link to="/">
              <Text>HOME</Text>
            </Link>
            <Link to="/admin/hotels">
              <Text>HOTELS</Text>
            </Link>
          </React.Fragment>
        ) : (
          <React.Fragment>
            <Link to="/">
              <Text>HOME</Text>
            </Link>
            <Link to="/messages">
              <Text>MESSAGES</Text>
            </Link>
            <Link to="/calendar">
              <Text>CALENDAR</Text>
            </Link>
            <Link to="/proposals">
              <Text>PROPOSALS</Text>
            </Link>
            <Link to="/bookings">
              <Text>BOOKINGS</Text>
            </Link>
            <Link to="/holds">
              <Text>HOLDS</Text>
            </Link>
          </React.Fragment>
        )}
        <UserDropdown user={currentUser} />
      </Links>
    </Content>
  </Container>
);

const mapStateToProps = state => ({
  currentUser: getCurrentUser(state),
});

export default connect(mapStateToProps)(Header);
