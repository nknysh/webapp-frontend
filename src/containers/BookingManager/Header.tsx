import React from 'react';
import styled from 'styled-components';
import { IBooking } from 'services/BackendApi';
import { IBookingLeadGuestInformation } from 'store/modules/bookingManager/model';
import { formatDateDisplay, currencyCodeToSymbol, formatPrice } from 'utils';
import { pureUiTheme } from 'pureUi/pureUiTheme';
import { IconButton } from 'pureUi/Buttons';
import FileCopyIcon from '@material-ui/icons/FileCopy';
import copy from 'copy-to-clipboard';

export interface IHeaderComponentProps {
  className?: string;
  booking: IBooking;
  leadGuestInformation: IBookingLeadGuestInformation;
  compactGuestBreakdown: string;
}
const HeaderComponent = (props: IHeaderComponentProps) => {
  if (!props.booking.breakdown) {
    return null;
  }

  const { total, totalBeforeDiscount } = props.booking.breakdown.totals;

  return (
    <div className={props.className}>
      <div className="reference">
        <IconButton
          className="icon-button"
          onClick={() => {
            if (props.booking.humanReadableId) {
              copy(props.booking.humanReadableId)
            }
          }}
        >
          <FileCopyIcon />
        </IconButton>
        <span>Ref: #</span>
        <input readOnly id="booking-human-readable-id" type="text" value={props.booking.humanReadableId} />
      </div>

      <div className="booking-manager-header-inner">
        <div className="guest-details">
          <h1>
            {props.leadGuestInformation.guestTitle} {props.leadGuestInformation.guestFirstName}{' '}
            {props.leadGuestInformation.guestLastName}
          </h1>
          <h2>{props.booking.travelAgent?.companySignupInfo.name}</h2>
          <p>
            <label>Agent</label> {props.booking.travelAgent?.firstName} {props.booking.travelAgent?.lastName}
          </p>
          <p>
            <label>Email</label> {props.booking.travelAgent?.email}
          </p>
          <p>
            <label>Tel</label> {props.booking.travelAgent?.phoneNumber}
          </p>
        </div>

        <div className="resort-details">
          <h1>{props.booking.hotelName}</h1>
          <p>
            <label>Arrival</label>{' '}
            {props.booking.flightArrivalDate ? formatDateDisplay(props.booking.flightArrivalDate) : '/'}
          </p>
          <p>
            <label>Depart</label>{' '}
            {props.booking.flightDepartureDate ? formatDateDisplay(props.booking.flightDepartureDate) : '/'}
          </p>
          <p>
            <label>Guests</label> {props.compactGuestBreakdown}
          </p>
        </div>

        <div className="price-details">
          <label>Total</label>
          <h1>
            {currencyCodeToSymbol(props.booking.breakdown.currency)}
            {formatPrice(total)}
          </h1>
          <h2>
            Before Discount: {currencyCodeToSymbol(props.booking.breakdown.currency)}
            {formatPrice(totalBeforeDiscount)}
          </h2>
        </div>
      </div>
    </div>
  );
};

export const Header = styled(HeaderComponent)`
  width: 100%;
  margin: 0 auto;
  padding: 20px 10px;

  .reference {
    display: flex;
    align-items: center;
  }

  .reference span, .reference input {
    font-size: ${pureUiTheme.typography.sizes.default}px;
  }

  .reference .icon-button {
    padding-left: 0;
    color: ${pureUiTheme.colors.teal};
    cursor: pointer;
  }

  .reference input {
    background: none;
    border: none;
    outline: none;
  }

  .booking-manager-header-inner {
    display: flex;
    justify-content: space-between;
  }

  h1 {
    font-size: ${pureUiTheme.typography.sizes.big}px;
  }
  h2 {
    font-size: ${pureUiTheme.typography.sizes.mid}px;
  }
  label,
  p {
    font-size: ${pureUiTheme.typography.sizes.default}px;
  }

  p > label {
    color: ${pureUiTheme.colors.grayDarker};
  }

  .guest-details {
    margin-right: 40px;
  }

  .guest-details h1 {
    font-family: ${pureUiTheme.typography.serifFont};
  }

  .guest-details h2 {
    font-weight: normal;
    margin 8px 0px;
    font-size: ${pureUiTheme.typography.sizes.less}px;
  }

  .resort-details h1 {
    font-family: ${pureUiTheme.typography.serifFont};
  }

  .resort-details {
    flex-grow: 2;
  }

  .price-details {
    text-align: right;
  }

  .price-details h1 {
    font-size: ${pureUiTheme.typography.sizes.bigger}px;
    color: ${pureUiTheme.colors.green};
  }

  .price-details h2 {
    color: ${pureUiTheme.colors.grayDarker};
    font-weight: normal;
  }

  @media (max-width: 600px) {
    .booking-manager-header-inner {
      flex-wrap: wrap;
    }

    .guest-details {
      width: 100%;
      margin-right: 0px;
    }

    .resort-details {
      width: 100%;
    }

    .price-details {
      width: 100%;
      text-align: left;
    }
  }
`;
