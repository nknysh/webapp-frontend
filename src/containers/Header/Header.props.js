import PropTypes from 'prop-types';
import { propOr } from 'ramda';

import { propTypes as menuPropTypes } from 'components/elements/Menu';
import { propTypes as withAuthPropTypes } from 'hoc/withAuthentication';

export const propTypes = {
  ...withAuthPropTypes,
  menu: propOr(PropTypes.any, 'links', menuPropTypes),
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
};

export const defaultProps = {
  location: {
    pathname: '',
  },
};

export const contextTypes = Object.freeze({
  SIGN_UP: 'signup',
  LOGIN: 'login',
});
