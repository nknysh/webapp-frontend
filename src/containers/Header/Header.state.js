import { connect } from 'react-redux';

import { getUiHeaderMenu } from 'store/modules/ui';
import { getLoggedIn, isSR } from 'store/modules/auth';

export const mapStateToProps = state => ({
  isSR: isSR(state),
  menu: getUiHeaderMenu(state),
  loggedIn: getLoggedIn(state),
});

export default connect(mapStateToProps);
