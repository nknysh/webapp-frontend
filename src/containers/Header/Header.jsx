import React from 'react';
import { compose } from 'ramda';

import { Menu } from 'components/Menu';
import { Link } from 'components/Link';
import { withAuthentication } from 'hoc/withAuthentication';

import logo from './assets/header-logo.png';

import { propTypes } from './Header.props';
import connect from './Header.state';
import { HeaderContainer, StyledHeader } from './Header.styles';

export const Header = ({ menu, className }) => (
  <StyledHeader className={className}>
    <HeaderContainer>
      <Link to="/">{logo && <img src={logo} />}</Link>
      <Menu align="end" links={menu} />
    </HeaderContainer>
  </StyledHeader>
);

Header.propTypes = propTypes;

export default compose(
  withAuthentication,
  connect
)(Header);
