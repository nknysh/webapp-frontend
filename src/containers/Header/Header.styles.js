import styled, { css } from 'styled-components';
import { Icon } from '@material-ui/core';

import { Container } from 'styles/elements';
import breakpoints from 'styles/breakpoints';
import theme from 'styles/theme';

import { Link, Menu } from 'components';

export const HeaderContainer = styled(Container)`
  display: flex;
  width: 100%;
  height: ${theme.headerSizes.mobile}${theme.unit};
  align-items: center;

  ${breakpoints.tablet`
    height: ${theme.headerSizes.tablet}${theme.unit};
  `}
`;

export const StyledHeader = styled.div`
  background: ${theme.backgroundColor}
  border-bottom: 1${theme.unit} solid ${theme.colors['gray-light']};
  padding: ${theme.gutter}${theme.unit} ${theme.gutter * 2}${theme.unit};

  ${breakpoints.tablet`
    padding: ${theme.gutter};
  `}
`;

export const HeaderLogo = styled(Link)`
  padding: ${theme.gutter}${theme.unit};
  height: 26${theme.unit};
  width: auto;
  display: block;

  a{
    display: block;
  }
    
  img {
    height: 100%;
    width: auto;
  }
  ${breakpoints.tablet`
    height: auto;
  `}
`;

export const HeaderMenuArea = styled.div`
  width: 100%;
  text-align: right;
`;

export const HeaderMobileMenuButton = styled(Icon)`
  display: block;
  color: ${theme.colors.gold};
  font-size: 38${theme.unit} !important;

  ${breakpoints.tablet`
    display: none !important;
  `}
`;

export const HeaderMenu = styled(Menu)`
  position: absolute;
  background: ${theme.backgroundColor};
  padding: ${theme.gutter}${theme.unit};
  top: ${theme.headerSizes.mobile + theme.gutter * 2}${theme.unit};
  right: 0;
  left: 0;
  border-top: 1${theme.unit} solid ${theme.colors.gold};
  transition: ease-in-out 0.25s all;
  text-align: left;
  font-size: 12${theme.unit};

  ${({ isOpen }) => css`
    opacity: ${isOpen ? 1 : 0};
    pointer-events: ${isOpen ? 'auto' : 'none'};
  `}

  ${breakpoints.tablet`
    border: 0;
    position: relative;
    padding: 0;
    top: unset;
    right: unset;
    left: unset;
    width: 100%;
    opacity: 1;
    pointer-events: auto;
    font-size: 11${theme.unit};
  `}
`;

export const HeaderMenuWrapper = styled.div``;
