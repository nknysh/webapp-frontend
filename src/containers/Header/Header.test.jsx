import React from 'react';

import headerLinks from 'config/links/header--logged-out';

import { Header } from './Header';

const getComponent = props => shallow(<Header menu={headerLinks} isAuthenticated={false} {...props} />);

describe('<Header />', () => {
  describe('render', () => {
    it('matches logged out snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
    it('matches logged in snapshot', () => {
      expect(getComponent({ isAuthenticated: true })).toMatchSnapshot();
    });
  });
});
