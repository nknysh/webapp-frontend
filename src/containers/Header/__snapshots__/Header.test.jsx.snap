// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render matches logged in snapshot 1`] = `
<styled.div>
  <Styled(styled.div)>
    <Styled(Link)
      to="/"
    />
    <styled.div>
      <Styled(WithStyles(Icon))
        onClick={[Function]}
      >
        menu
      </Styled(WithStyles(Icon))>
      <Styled(Menu)
        align="end"
        isOpen={false}
        links={
          Object {
            "createAccount": Object {
              "href": "/sign-up",
              "title": "Create an account",
            },
          }
        }
        onLinkClick={[Function]}
      />
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;

exports[`<Header /> render matches logged out snapshot 1`] = `
<styled.div>
  <Styled(styled.div)>
    <Styled(Link)
      to="/"
    />
    <styled.div>
      <Styled(WithStyles(Icon))
        onClick={[Function]}
      >
        menu
      </Styled(WithStyles(Icon))>
      <Styled(Menu)
        align="end"
        isOpen={false}
        links={
          Array [
            Object {
              "data-active": false,
              "href": "",
              "onClick": [Function],
              "title": "Create an account",
            },
            Object {
              "data-active": false,
              "href": "",
              "onClick": [Function],
            },
          ]
        }
        onLinkClick={[Function]}
      />
    </styled.div>
  </Styled(styled.div)>
  <Modal
    onBackdropClick={[Function]}
    onClose={[Function]}
    onEscapeKeyDown={[Function]}
    open={false}
  />
</styled.div>
`;
